src/wallet/wallet.cpp

```diff
@@ -1699,6 +1699,12 @@ boost::optional<uint256> CWallet::GetSproutNoteNullifier(const JSDescription &js
         hSig,
         (unsigned char) n);
     auto note = note_pt.note(address);
+
+    // Check note plaintext against note commitment
+    if (note.cm() != jsdesc.commitments[n]) {
+        throw libzcash::note_decryption_failed();
+    }
```

Comment:
Non-blocking: this should be moved into `libzcash::SproutNotePlaintext::decrypt()`, to match what we do for Sapling.

Reply:
Please file another ticket for this.